File: Semester1/labs/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int wordCount(string sentence);
int letterCount(string sentence);
int sentenceCount(string sentence);

int main(void)
{
    string text = get_string("Your Sentence: ");
    int letters = letterCount(text);
    int words = wordCount(text);
    int sentences = sentenceCount(text);
    float index = 0.0588 * ((float) (letters) *100 / words) - 0.296 * ((float) sentences * 100 / words) - 15.8;
    if ((int) round(index) > 16)
    {
        printf("Grade 16+\n");
    }
    else if ((int) round(index) < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

int wordCount(string sentence)
{
    int wordCount = 1;
    for (int i = 0; i < strlen(sentence); i++)
    {
        // checks for spaces and adds to word count
        if (sentence[i] == ' ')
        {
            wordCount += 1;
        }
    }
    return wordCount;
}

int letterCount(string sentence)
{
    int letterCount = 0;
    for (int i = 0; i < strlen(sentence); i++)
    {
        // checks if charater is uppercase or losecase letter then adds to letter count
        if ((sentence[i] >= 65 && sentence[i] <= 90) || (sentence[i] >= 97 && sentence[i] <= 122))
        {
            letterCount += 1;
        }
    }
    return letterCount;
}

int sentenceCount(string sentence)
{
    int sentenceCount = 0;
    for (int i = 0; i < strlen(sentence); i++)
    {
        // checks for punctuations and adding to sentence count
        if (sentence[i] == '!' || sentence[i] == '.' || sentence[i] == '?')
        {
            sentenceCount += 1;
        }
    }
    return sentenceCount;
}

File: Semester2/labs/readability.py
'''
Derek Yee
Predicts the age of a piece of writing based on the Coleman-Liau formula
'''


def main():
    # gets text input
    text = input("Text: ").strip()
    # counts sentences, words and letters
    sentences = sentenceCount(text)
    words = wordCount(text)
    letters = characterCount(text)
    # calculates based on Coleman-Liau formula
    index = 0.0588 * (letters * 100 / words) - 0.296 * (sentences * 100 / words) - 15.8
    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(index)}")


def sentenceCount(input):
    sentenceCount = 0
    punctuations = ["?", "!", "."]
    for i in input:
        if i in punctuations:
            sentenceCount += 1
    return sentenceCount


def wordCount(input):
    # one more word than number of spaces in sentence
    wordCount = 1
    for i in input:
        if i == " ":
            wordCount += 1
    return wordCount


def characterCount(input):
    characterCount = 0
    for i in input:
        if i.isalpha():
            characterCount += 1
    return characterCount


main()
